### PR TITLE
Correct conversions errors from u64 to size_t(u32 on msvc32)

### LINKED ISSUE #91: MSVC 2017 build fails with x86 architecture (v2.0.0-rc2)
Using MSVC 2017 Community on Windows 10 and CMake 3.11.3, I am trying to build Windows binary from  jumanpp-2.0.0-rc2.tar.xz. However, I have some trouble to ask the developers for help. (On Linux, I got working binary from the same source. Many thanks!)

Procedure
1. Install MSVC 2017 Comunnity version (chose default set for Desktop development with C++) and CMake 3.11.3 (chose Windows win64-x64 ZIP as the installer gave an error).
jumanpp01
jumanpp02
jumanpp05
jumanpp03
2. Download jumanpp-2.0.0-rc2.tar.xz and extract into C:\data\jumanpp-2.0.0-rc2 (I used 7-Zip 18.05).
3. As instructed https://github.com/ku-nlp/jumanpp#building-from-a-package , open a terminal and do the following with the result.
jumanpp04
```
C:\data\jumanpp-2.0.0-rc2>mkdir bld

C:\data\jumanpp-2.0.0-rc2>cd bld

C:\data\jumanpp-2.0.0-rc2\bld>cmake ..  -DCMAKE_BUILD_TYPE=Release    -DCMAKE_INSTALL_PREFIX="C:\Program Files\jumanpp"
-- Building for: Visual Studio 15 2017
-- Selecting Windows SDK version 10.0.17134.0 to target Windows 10.0.16299.
-- The C compiler identification is MSVC 19.14.26430.0
-- The CXX compiler identification is MSVC 19.14.26430.0
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.14.26428/bin/Hostx86/x86/cl.exe
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.14.26428/bin/Hostx86/x86/cl.exe -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.14.26428/bin/Hostx86/x86/cl.exe
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.14.26428/bin/Hostx86/x86/cl.exe -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for pthread.h
-- Looking for pthread.h - not found
-- Found Threads: TRUE
-- Could NOT find Protobuf (missing: Protobuf_LIBRARIES Protobuf_INCLUDE_DIR)
Not using Protobuf-based components
-- Eigen3... OK

    Static feature generation:
        COMMAND cgtest02_codegen_binary cgtest02 Test02 C:/data/jumanpp-2.0.0-rc2/bld/src/core/codegen/gen
        DEPENDS cgtest02_codegen_binary cg_2_spec.h
        OUTPUT C:/data/jumanpp-2.0.0-rc2/bld/src/core/codegen/gen/cgtest02.cc C:/data/jumanpp-2.0.0-rc2/bld/src/core/codegen/gen/cgtest02.h
        LIBS jpp_core


    Static feature generation:
        COMMAND jpp_jumandic_cg_codegen_binary jpp_jumandic_cg JumandicStatic C:/data/jumanpp-2.0.0-rc2/bld/src/jumandic/gen
        DEPENDS jpp_jumandic_cg_codegen_binary shared/jumandic_spec.cc
        OUTPUT C:/data/jumanpp-2.0.0-rc2/bld/src/jumandic/gen/jpp_jumandic_cg.cc C:/data/jumanpp-2.0.0-rc2/bld/src/jumandic/gen/jpp_jumandic_cg.h
        LIBS jpp_jumandic_spec

C:/data/jumanpp-2.0.0-rc2/bld/src/jumandic/gen
-- Configuring done
-- Generating done
-- Build files have been written to: C:/data/jumanpp-2.0.0-rc2/bld
```
4. Doubleclick "C:\data\jumanpp-2.0.0-rc2\bld\jumanpp.sln" and launch MSVC 2017.
5. After some messages and project loading is done, choose ALL_BUILD from Solution Explorer and build with solution configuration of Release (changed from Debug) and Win32 (as default).
jumanpp06
6. After a while, I get 19 errors and many warnings and the build fails. Attached gz file is the exact output of the build (1210 lines long).
jumanpp07
[jumanpp-build-log.txt.gz](https://github.com/ku-nlp/jumanpp/files/2076480/jumanpp-build-log.txt.gz)

Since Linux build is fine and it actually works pretty well, something on my Windows is not right. Some advice would be really appreciated.

### PR DESCRIPTION
**Pull request by DoumanAsh**
There are still lots of warnings due to conversions between 32/64 bit integers, but I fixed only errors and these warnings around them.

Users should be able build it with msvc32 compiler

Closes  #91

### CODE REVIEW

File: src/rnn/mikolov_rnn.cc [hunk 1]
```diff
@@ -177,29 +177,33 @@ Status MikolovModelReader::parse() {
...
                                  hdr.layerSize * hdr.layerSize});
   // 3 comes from rounding to next value + sizeof(float)
-  auto pageSize = 1ULL << (static_cast<u32>(std::log2(maxBlock)) + 3);
+  auto pageSize = static_cast<size_t>(1) << (static_cast<u32>(std::log2(maxBlock)) + 3);
   data_->memmgr.initialize(pageSize);
   data_->alloc = data_->memmgr.value().core();
   auto& alloc = data_->alloc;
+
+  const auto embedding_size = static_cast<size_t>(hdr.layerSize) * static_cast<size_t>(hdr.vocabSize);
+  const auto matrix_size = static_cast<size_t>(hdr.layerSize) * static_cast<size_t>(hdr.layerSize);
```
Comment (DoumanAsh) [inline src/rnn/mikolov_rnn.cc]: I had second thought about whether `hdr.layerSize * hdr.layerSize` is mistake or not...

Comment (eiennohito) [reply src/rnn/mikolov_rnn.cc]: It's a square matrix layerSize times layerSize, so that's correct

File: src/rnn/mikolov_rnn.cc [hunk 2]
```diff
@@ -177,29 +177,33 @@ Status MikolovModelReader::parse() {
   auto maxBlock = std::max<u64>({hdr.layerSize * hdr.vocabSize, hdr.maxentSize,
                                  hdr.layerSize * hdr.layerSize});
   // 3 comes from rounding to next value + sizeof(float)
-  auto pageSize = 1ULL << (static_cast<u32>(std::log2(maxBlock)) + 3);
+  auto pageSize = static_cast<size_t>(1) << (static_cast<u32>(std::log2(maxBlock)) + 3);
   data_->memmgr.initialize(pageSize);
   data_->alloc = data_->memmgr.value().core();
   auto& alloc = data_->alloc;
+
+  const auto embedding_size = static_cast<size_t>(hdr.layerSize) * static_cast<size_t>(hdr.vocabSize);
```
Comment (eiennohito) [inline src/rnn/mikolov_rnn.cc]: Could you name this embedding_matrix_size
Usually an embedding for a single token is called embedding.

Next thing should be called transition_matrix_size in that case.

Comment (DoumanAsh) [reply src/rnn/mikolov_rnn.cc]: Renamed it

Comment (eiennohito) [reply src/rnn/mikolov_rnn.cc]: transition_matrix_size as well please!

Anyway, congratulations on 1000th commit :p

Comment (DoumanAsh) [reply src/rnn/mikolov_rnn.cc]: Oh sorry, I thought you meant it as if we leave name as it is

Comment (eiennohito) [reply src/rnn/mikolov_rnn.cc]: Sorry for the confusion.

Need to improve my clearness.

Comment (DoumanAsh) [reply src/rnn/mikolov_rnn.cc]: Addressed.

And thanks, sorry about ruining nice number `999` 😄

**Comment by eiennohito** [comment]
https://ci.appveyor.com/project/eiennohito/jumanpp/builds/20694567/job/u883gac1of6chfnj

Some test fail; I won't enable it by default.

**Comment by DoumanAsh** [comment]
Oh, let me re-check it, I remember I had no troubles yesterday

**Pull request merged**